providers/marketo/metadata.go

```diff
@@ -37,22 +38,57 @@ func (c *Connector) ListObjectMetadata(ctx context.Context,
 
 		resp, err := c.Client.Get(ctx, url.String())
 		if err != nil {
-			metadataResult.Errors[obj] = err
+			// Try fallback function
```

Comment:
* Since this fallback appears multiple times, you can put it in a helper and call that
* You can combine the checks as `err != nil || resp == nil...`